presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -107,6 +137,51 @@ public static double ceiling(@SqlType(StandardTypes.DOUBLE) double num)
         return Math.ceil(num);
     }
 
+    private static SqlScalarFunction decimalCeilingFunction(String name)
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name(name)
+                .literalParameters("num_precision", "num_scale", "return_precision")
+                .longVariableConstraints(longVariableCalculation("return_precision", "num_precision - num_scale + min(num_scale, 1)"))
+                .argumentTypes("decimal(num_precision, num_scale)")
+                .returnType("decimal(return_precision,0)")
+                .build();
+        return SqlScalarFunction.builder(MathFunctions.class)
+                .signature(signature)
+                .methods("ceilingShortDecimal", "ceilingLong2ShortDecimal", "ceilingLongDecimal")
+                .extraParameters(MathFunctions::decimalTenToScaleExtraParameters)
+                .build();
+    }
+
+    private static List<Object> decimalTenToScaleExtraParameters(SpecializeContext context)
+    {
+        return ImmutableList.of(tenToNth(context.getLiteral("num_scale").intValue()));
+    }
+
+    public static long ceilingShortDecimal(long num, BigInteger divisorBigInteger)
+    {
+        long divisor = divisorBigInteger.longValueExact();
+        long increment = (num % divisor) > 0 ? 1 : 0;
+        return num / divisor + increment;
+    }
+
+    public static long ceilingLong2ShortDecimal(Slice num, BigInteger divisor)
+    {
+        return ceiling(num, divisor).longValueExact();
+    }
+
+    public static Slice ceilingLongDecimal(Slice num, BigInteger divisor)
+    {
+        return LongDecimalType.unscaledValueToSlice(ceiling(num, divisor));
+    }
+
+    private static BigInteger ceiling(Slice num, BigInteger divisor)
+    {
+        BigInteger[] divideAndRemainder = unscaledValueToBigInteger(num).divideAndRemainder(divisor);
+        return divideAndRemainder[0].add(BigInteger.valueOf(divideAndRemainder[0].signum() > 0 ? 1 : 0));
```

Comment:
this is wrong. `ceil(333333333333333333333333.0)` will be calculated to `333333333333333333333334.0`. Fix and add tests.